src/opensles/EngineOpenSLES.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "common/OboeDebug.h"
+#include "EngineOpenSLES.h"
+#include "OpenSLESUtilities.h"
+
+using namespace oboe;
+
+EngineOpenSLES *EngineOpenSLES::getInstance() {
+    static EngineOpenSLES sInstance;
+    return &sInstance;
```

Comment:
It is preferred to return a reference (EngineOpenSLES&) rather than pointer for preventing clients from attempting to calling 'delete' on it or putting it into a smart pointer.

Also, references can't be null which makes caller's life easier.

Reply:
Done